### PR TITLE
Fix #179: Remove stale build label when the pull request gets updated

### LINKED ISSUE #179: Bot should remove the "Stale Build" label once a new commit is pushed
Currently, the Oppia bot adds a "Stale Build" label to the PRs where the PR CI builds ran 2+ days ago. But even if someone pushes new changes to the PR the Oppia bot doesn't remove the "Stale build" label.

Through this issue, we are expecting to remove the "Stale Build" label once a PR gets updated.

### PR DESCRIPTION
**Pull request by jameesjohn**
## Explanation

Fixed #179: Removes the stale build label when the pull request gets updated by the author.
## Checklist
- [x] I have successfully deployed my own instance of Oppiabot.
  - You can find instructions for doing this [here](https://github.com/oppia/oppiabot/wiki/Deploying-your-own-instance-of-the-oppiabot).
- [x] I have manually tested all the changes made in this PR following the [manual tests matrix](https://github.com/oppia/oppiabot/wiki/Manual-Tests-Matrix).

### CODE REVIEW

File: constants.js [hunk 1]
```diff
@@ -86,7 +86,8 @@ const checksWhitelist = {
       mergeConflictCheck,
       jobCheck,
       modelCheck,
-      codeOwnerCheck
+      codeOwnerCheck,
+      oldBuildLabelCheck
```
Comment (vojtechjelinek) [inline constants.js:90]: Remove this from `[pushEvent]` and maybe also add `forcePushCheck` here? The second part can be done in different PR.

File: lib/staleBuildChecks.js [hunk 2]
```diff
@@ -91,6 +64,21 @@ const removeOldBuildLabel = async (context) => {
...
+    context.repo({
+      ref: pullRequest.head.sha
+    })
+  );
+
+  const lastCommitDate = new Date(lastCommit.commit.author.date);
+  // Commit is older than 2 days
+  if (utils.MIN_BUILD_DATE > lastCommitDate) {
+    // Will be called if the synchronize event got triggered as a
+    // result of a merge to the main branch.
```
Comment (vojtechjelinek) [inline lib/staleBuildChecks.js:78]:
```suggestion
```

File: lib/staleBuildChecks.js [hunk 3]
```diff
@@ -91,6 +64,21 @@ const removeOldBuildLabel = async (context) => {
...
 
+  // Get last commit for pull request.
+  const {data: lastCommit} = await context.github.repos.getCommit(
+    context.repo({
+      ref: pullRequest.head.sha
+    })
+  );
+
+  const lastCommitDate = new Date(lastCommit.commit.author.date);
+  // Commit is older than 2 days
```
Comment (vojtechjelinek) [inline lib/staleBuildChecks.js:75]:
```suggestion
  // Commit is older than 2 days, this can happen when 'synchronize' event 
  // got triggered as a result of a merge to the main branch. In this case
  // we don't want to remove the label.
```

**Review by vojtechjelinek — CHANGES_REQUESTED** [review]
Thanks! just a few comments, did you test this properly?

File: constants.js (hunk 1 above)
Comment (jameesjohn) [reply constants.js:90]: Done.

File: index.js [hunk 4]
```diff
@@ -58,88 +58,108 @@ const runChecks = async (context, checkEvent) => {
...
+            );
             break;
           case constants.oldBuildLabelCheck:
-            await staleBuildModule.removeOldBuildLabel(context);
+            callable.push(staleBuildModule.removeOldBuildLabel(context));
             break;
         }
       }
+      // Wait for all checks to resolve or reject.
+      await Promise.allSettled(callable);
```
Comment (jameesjohn) [inline index.js:162]: This was needed because without this, any failure from a feature would prevent other features from running. This way, a feature's failure does not stop other features from running.

**Comment by jameesjohn** [comment]
> Thanks! just a few comments, did you test this properly?

Yes, this has been tested see https://github.com/jameesjohn/oppia/pull/39

File: .prettierrc [hunk 5]
```diff
@@ -0,0 +1 @@
+{"singleQuote": true}
```
Comment (vojtechjelinek) [inline .prettierrc:1]: Why was this added?

**Review by vojtechjelinek — CHANGES_REQUESTED** [review]
One small note.

File: .prettierrc (hunk 5 above)
Comment (jameesjohn) [reply .prettierrc]: It was to help with formatting. 
I'll add it to `.gitignore`

**Review by vojtechjelinek — APPROVED** [review]
LGTM!

**Comment by vojtechjelinek** [comment]
@jameesjohn Thanks! Can you release this, please?

**Comment by jameesjohn** [comment]
Done.

**Comment by SAEb-ai** [comment]
@vojtechjelinek @jameesjohn I think this is still occuring . See(https://github.com/oppia/oppia/pull/12370)
